composer/models/huggingface.py

```diff
@@ -286,13 +286,11 @@ def loss(self, outputs, batch):
             return outputs[0]
 
     def eval_forward(self, batch, outputs: Optional[Any] = None):
-        output = outputs if outputs else self.forward(batch)
-        if self.use_logits or batch.get('mode', None) == 'lm_task':
+        if self.use_logits or batch.get('mode', None) == 'icl_task':
+            # pop labels first to avoid computing loss
             self.labels = batch.pop('labels')
-            if self.shift_labels:
-                # HF CausalLM models internally shift labels before computing loss, so we do the same here
-                self.labels[:, :-1] = self.labels[:, 1:].clone()
-                self.labels[:, -1] = -100
+            output = outputs if outputs else self.forward(batch)
```

Comment:
I think this might actually cause a problem. If you pass the batch through without "labels", the logits are no longer at index one in the tuple. Still verifying, just want to block merging until we figure it out.

Reply:
ok, so we need to change
```
            if self.config.use_return_dict:
                output = output['logits']
            else:
                # logits are at index 1 in the output tuple
                output = output[1]
```
to
```
            if self.config.use_return_dict:
                output = output['logits']
            else:
                # logits are at index 0 because loss is not present
                output = output[0]
```